packages/nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -366,14 +365,15 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
         // INTERACTIONS
         // *******
 
-        for (uint256 j = 0; j < payouts.length; j++) {
-            if (payouts[j] > 0) {
-                bytes32 destination = allocation[indices.length > 0 ? indices[j] : j].destination;
+        for (uint256 j = 0; j < newAllocation.length; j++) {
```

Comment:
note that you can also iterate over `indices[j]` to save some gas when `indices.length` is small and `allocation.length` is large

Reply:
I tried this out: but because of needing to switch on the `indices = []` case, the code got significantly messier and ended up costing more gas in at least some scenarios.

I would like to punt this for now, but happy to circle back to it in future.